src/pages/OnboardingWorkspaces/BaseOnboardingWorkspaces.tsx

```diff
@@ -0,0 +1,159 @@
+            Report.completeOnboarding(
+                CONST.ONBOARDING_CHOICES.LOOKING_AROUND,
+                CONST.ONBOARDING_MESSAGES[CONST.ONBOARDING_CHOICES.LOOKING_AROUND],
+                onboardingPersonalDetails?.firstName ?? '',
+                onboardingPersonalDetails?.lastName ?? '',
+            );
+            Welcome.setOnboardingAdminsChatReportID();
+            Welcome.setOnboardingPolicyID(policyID);
+
+            navigateAfterOnboarding(isSmallScreenWidth, shouldUseNarrowLayout, canUseDefaultRooms, policyID, activeWorkspaceID, route.params?.backTo);
+        },
+        [onboardingPersonalDetails?.firstName, onboardingPersonalDetails?.lastName, isSmallScreenWidth, shouldUseNarrowLayout, canUseDefaultRooms, activeWorkspaceID, route.params?.backTo],
+    );
+    const policyIDItems = useMemo(() => {
+        return Object.values(joinablePolicies ?? {}).map((policyInfo) => {
+            return {
+                text: policyInfo.policyName,
+                alternateText: translate('onboarding.workspaceMemberList', {employeeCount: policyInfo.employeeCount, policyOwner: policyInfo.policyOwner}),
+                keyForList: policyInfo.policyID,
+                isDisabled: true,
+                rightElement: (
+                    <Button
+                        isDisabled={isOffline}
+                        success
+                        medium
+                        text={policyInfo.automaticJoiningEnabled ? translate('workspace.workspaceList.joinNow') : translate('workspace.workspaceList.askToJoin')}
+                        onPress={() => {
+                            handleJoinWorkspace(policyInfo.policyID);
+                        }}
+                    />
+                ),
+                icons: [
+                    {
+                        id: policyInfo.policyID,
+                        source: ReportUtils.getDefaultWorkspaceAvatar(policyInfo.policyName),
+                        fallbackIcon: Expensicons.FallbackWorkspaceAvatar,
+                        name: policyInfo.policyName,
+                        type: CONST.ICON_TYPE_WORKSPACE,
+                    },
+                ],
+            };
+        });
+    }, [translate, isOffline, joinablePolicies, handleJoinWorkspace]);
+
+    const wrapperPadding = onboardingIsMediumOrLargerScreenWidth ? styles.mh8 : styles.mh5;
+
+    useEffect(() => {
+        if (joinablePoliciesLoading === true || joinablePoliciesLoading === undefined || !joinablePolicies || Object.keys(joinablePolicies).length > 0) {
+            return;
+        }
+
+        Navigation.navigate(ROUTES.ONBOARDING_PURPOSE.getRoute(route.params?.backTo));
+    }, [joinablePoliciesLoading, joinablePolicies, route.params?.backTo]);
+
+    const handleBackButtonPress = useCallback(() => {
+        if (isValidated) {
+            Navigation.navigate(ROUTES.ONBOARDING_PERSONAL_DETAILS.getRoute());
+            return;
+        }
+        Navigation.goBack();
```

Comment:
How do we even get here if we are not validated? 🤔

Reply:
Pretty easily. The user presses the back button on the two factor authentication page.